// @uibakery/kit/esm2015/lib/components/bubble-map/bubble-map.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { HttpClient } from '@angular/common/http';
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input } from '@angular/core';
import { NbThemeService } from '@nebular/theme';
import { registerMap } from 'echarts';
import { combineLatest, Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
export class BkBubbleMapComponent {
    /**
     * @param {?} theme
     * @param {?} http
     * @param {?} cd
     */
    constructor(theme, http, cd) {
        this.theme = theme;
        this.http = http;
        this.cd = cd;
        this.options = {};
        this._data = { countryCoordinates: [], mapData: [] };
        this.colorConfig = {};
        this.geoColors = [];
        this.countryColorMap = {};
        this.max = -Infinity;
        this.destroyed$ = new Subject();
    }
    /**
     * @param {?} data
     * @return {?}
     */
    set data(data) {
        this._data = data;
        // avoid showing map without colors
        if (this.initialized) {
            this.updateOptions();
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        combineLatest([
            this.http.get('assets/map/world.json'),
            this.theme.getJsTheme(),
        ])
            .pipe(takeUntil(this.destroyed$))
            .subscribe((/**
         * @param {?} __0
         * @return {?}
         */
        ([map, config]) => {
            registerMap('world', map);
            this.initColors(config);
            this.updateOptions();
            this.initialized = true;
            this.cd.detectChanges();
        }));
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.destroyed$.next();
    }
    /**
     * @private
     * @param {?} config
     * @return {?}
     */
    initColors(config) {
        this.colorConfig = config.variables.bubbleMap;
        this.geoColors = [
            this.colorConfig.primary,
            this.colorConfig.info,
            this.colorConfig.success,
            this.colorConfig.warning,
            this.colorConfig.danger,
        ];
        this._data.mapData.forEach((/**
         * @param {?} itemOpt
         * @return {?}
         */
        itemOpt => {
            this.countryColorMap[itemOpt.name] = this.getRandomGeoColor();
        }));
    }
    /**
     * @private
     * @return {?}
     */
    updateOptions() {
        this._data.mapData.forEach((/**
         * @param {?} itemOpt
         * @return {?}
         */
        (itemOpt) => {
            if (itemOpt.value > this.max) {
                this.max = itemOpt.value;
            }
        }));
        this.options = {
            title: {
                text: 'World Population (2011)',
                left: 'center',
                top: 'top',
                textStyle: {
                    color: this.colorConfig.titleColor,
                },
            },
            tooltip: {
                trigger: 'item',
                formatter: (/**
                 * @param {?} params
                 * @return {?}
                 */
                params => {
                    return params.name + ': ' + params.value[2];
                }),
            },
            visualMap: {
                show: false,
                min: 0,
                max: this.max,
                inRange: {
                    symbolSize: [6, 60],
                },
            },
            geo: {
                name: 'World Population (2010)',
                type: 'map',
                map: 'world',
                roam: true,
                label: {
                    emphasis: {
                        show: false,
                    },
                },
                itemStyle: {
                    normal: {
                        areaColor: this.colorConfig.areaColor,
                        borderColor: this.colorConfig.areaBorderColor,
                    },
                    emphasis: {
                        areaColor: this.colorConfig.areaHoverColor,
                    },
                },
                zoom: 1.1,
            },
            series: [
                {
                    type: 'scatter',
                    coordinateSystem: 'geo',
                    data: this._data.mapData.map((/**
                     * @param {?} itemOpt
                     * @return {?}
                     */
                    itemOpt => {
                        return {
                            name: itemOpt.name,
                            value: [
                                this._data.countryCoordinates[itemOpt.code].longitude,
                                this._data.countryCoordinates[itemOpt.code].latitude,
                                itemOpt.value,
                            ],
                            itemStyle: {
                                normal: {
                                    color: this.getCountryColor(itemOpt.name),
                                },
                            },
                        };
                    })),
                },
            ],
        };
    }
    /**
     * @private
     * @param {?} countryName
     * @return {?}
     */
    getCountryColor(countryName) {
        /** @type {?} */
        const color = this.countryColorMap[countryName];
        if (color) {
            return color;
        }
        /** @type {?} */
        const generatedColor = this.getRandomGeoColor();
        this.countryColorMap[countryName] = generatedColor;
        return generatedColor;
    }
    /**
     * @private
     * @return {?}
     */
    getRandomGeoColor() {
        /** @type {?} */
        const index = Math.round(Math.random() * this.geoColors.length);
        return this.geoColors[index];
    }
}
BkBubbleMapComponent.decorators = [
    { type: Component, args: [{
                selector: 'bk-bubble-map',
                changeDetection: ChangeDetectionStrategy.OnPush,
                template: `
    <div echarts [options]="options" [autoResize]="true" [style.height.%]="100"></div>
  `,
                styles: [':host { display: block }']
            }] }
];
/** @nocollapse */
BkBubbleMapComponent.ctorParameters = () => [
    { type: NbThemeService },
    { type: HttpClient },
    { type: ChangeDetectorRef }
];
BkBubbleMapComponent.propDecorators = {
    data: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    BkBubbleMapComponent.prototype.options;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype._data;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.colorConfig;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.geoColors;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.countryColorMap;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.max;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.initialized;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.destroyed$;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.theme;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.http;
    /**
     * @type {?}
     * @private
     */
    BkBubbleMapComponent.prototype.cd;
}
