// @uibakery/kit/esm5/lib/components/map/map.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as leaflet from 'leaflet';
var BkMapComponent = /** @class */ (function () {
    function BkMapComponent() {
        this.options = {
            layers: [
                leaflet.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', { maxZoom: 18, attribution: '...' }),
            ],
            zoom: 5,
            center: leaflet.latLng({ lat: 38.991709, lng: -76.886109 }),
        };
    }
    BkMapComponent.decorators = [
        { type: Component, args: [{
                    selector: 'bk-map',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    template: "\n    <div leaflet [leafletOptions]=\"options\"></div>\n  ",
                    styles: [":host{display:block}div[leaflet]{width:100%;height:100%}"]
                }] }
    ];
    return BkMapComponent;
}());
export { BkMapComponent };
if (false) {
    /** @type {?} */
    BkMapComponent.prototype.options;
}
