// @uibakery/kit/esm2015/lib/components/echarts/echarts.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectorRef, Directive, ElementRef, NgZone } from '@angular/core';
import { ɵa as NgxEchartsDirective } from 'ngx-echarts';
// tslint:disable-next-line:directive-selector
export class EchartsDirective extends NgxEchartsDirective {
    /**
     * @param {?} element
     * @param {?} cd
     * @param {?} ngZone
     */
    constructor(element, cd, ngZone) {
        super(element, ngZone);
        this.element = element;
        this.cd = cd;
        this.alive = true;
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.alive = false;
    }
    /**
     * @return {?}
     */
    ngDoCheck() {
        super.ngDoCheck();
        this.invalidate();
    }
    /**
     * Echarts work in a strange way, they're detecting if component visible or not
     * and only then trying to decide render chart or not.
     * In a number of cases like Nebular Tabset we have element already rendered but hidden for now.
     * And when we're trying to make it visible echarts doesn't react and stays not rendered.
     *
     * Invoking intersection observer on each doCheck provides us the ability to track
     * whether we need to update charts component or not.
     *
     * @private
     * @return {?}
     */
    invalidate() {
        /** @type {?} */
        const o = new IntersectionObserver((/**
         * @param {?} entries
         * @return {?}
         */
        (entries) => {
            const [entry] = entries;
            // Means component is visible
            if (entry.isIntersecting) {
                if (this.alive) {
                    this.cd.detectChanges();
                }
                o.disconnect();
            }
        }));
        o.observe(this.element.nativeElement);
    }
}
EchartsDirective.decorators = [
    { type: Directive, args: [{ selector: '[echarts]' },] }
];
/** @nocollapse */
EchartsDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef },
    { type: NgZone }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    EchartsDirective.prototype.alive;
    /**
     * @type {?}
     * @private
     */
    EchartsDirective.prototype.element;
    /**
     * @type {?}
     * @private
     */
    EchartsDirective.prototype.cd;
}
