// @uibakery/kit/esm5/lib/components/line-chart/line-chart.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { NbThemeService } from '@nebular/theme';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';
var BkLineChartComponent = /** @class */ (function () {
    function BkLineChartComponent(theme) {
        this.theme = theme;
        this.options = {};
        this._data = { income: [], outcome: [], profit: [] };
        this.colorConfig = {};
        this.destroyed$ = new Subject();
    }
    Object.defineProperty(BkLineChartComponent.prototype, "data", {
        set: /**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            this._data = data;
            this.updateOptions();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    BkLineChartComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.theme.getJsTheme()
            .pipe(takeUntil(this.destroyed$))
            .subscribe((/**
         * @param {?} config
         * @return {?}
         */
        function (config) {
            _this.colorConfig = config.variables.charts;
            _this.updateOptions();
        }));
    };
    /**
     * @return {?}
     */
    BkLineChartComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.destroyed$.next();
    };
    /**
     * @private
     * @return {?}
     */
    BkLineChartComponent.prototype.updateOptions = /**
     * @private
     * @return {?}
     */
    function () {
        this.options = {
            backgroundColor: this.colorConfig.bg,
            color: [this.colorConfig.danger, this.colorConfig.primary, this.colorConfig.info],
            tooltip: {
                trigger: 'item',
                formatter: '{a} <br/>{b} : {c}',
            },
            xAxis: [
                {
                    type: 'category',
                    data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'],
                    axisTick: {
                        alignWithLabel: true,
                    },
                    axisLine: {
                        lineStyle: {
                            color: this.colorConfig.axisLineColor,
                        },
                    },
                    axisLabel: {
                        textStyle: {
                            color: this.colorConfig.textColor,
                        },
                    },
                },
            ],
            yAxis: [
                {
                    axisLine: {
                        lineStyle: {
                            color: this.colorConfig.axisLineColor,
                        },
                    },
                    splitLine: {
                        lineStyle: {
                            color: this.colorConfig.splitLineColor,
                        },
                    },
                    axisLabel: {
                        textStyle: {
                            color: this.colorConfig.textColor,
                        },
                    },
                },
            ],
            grid: {
                left: '3%',
                right: '4%',
                bottom: '3%',
                containLabel: true,
            },
            series: [
                {
                    name: 'Income',
                    type: 'line',
                    data: this._data.income,
                },
                {
                    name: 'Outcome',
                    type: 'line',
                    data: this._data.outcome,
                },
                {
                    name: 'Profit',
                    type: 'line',
                    data: this._data.profit,
                },
            ],
        };
    };
    BkLineChartComponent.decorators = [
        { type: Component, args: [{
                    selector: 'bk-line-chart',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    template: "\n    <div echarts [options]=\"options\" [autoResize]=\"true\" [style.height.%]=\"100\"></div>\n  ",
                    styles: [':host { display: block }']
                }] }
    ];
    /** @nocollapse */
    BkLineChartComponent.ctorParameters = function () { return [
        { type: NbThemeService }
    ]; };
    BkLineChartComponent.propDecorators = {
        data: [{ type: Input }]
    };
    return BkLineChartComponent;
}());
export { BkLineChartComponent };
if (false) {
    /** @type {?} */
    BkLineChartComponent.prototype.options;
    /**
     * @type {?}
     * @private
     */
    BkLineChartComponent.prototype._data;
    /**
     * @type {?}
     * @private
     */
    BkLineChartComponent.prototype.colorConfig;
    /**
     * @type {?}
     * @private
     */
    BkLineChartComponent.prototype.destroyed$;
    /**
     * @type {?}
     * @private
     */
    BkLineChartComponent.prototype.theme;
}
